### ISSUE TITLE
"Ignored names" from global settings are not handled correctly

### ISSUE THREAD

**Issue by mik01aj** [atom/find-and-replace#806]
My company uses a proprietary build system `adslkfsdjk`. It produces lots of files in directories named `adslkfsdjk-genfiles`, `adslkfsdjk-out`, `adslkfsdjk-testlogs`, etc. I added `adslkfsdjk-*` as ignored in Atom settings, and they disappeared from tree view. Good!

But when I search for something using `find-and-replace`, it seems to search all the files (including these), because it hangs forever showing an empty result list with this: 

![image](https://cloud.githubusercontent.com/assets/3027415/19929500/f21e6f84-a103-11e6-9a94-14ae277cf8aa.png)

...and my project has only around 2000 files. :open_mouth: 

Note that **when I put a list of these folder names explicitly in "Ignored names", the search starts working** (and it's reasonably fast). Seems like a bug to me. :bug:

**Comment by mik01aj** [comment]
One more: When I specify a file/directory pattern like `**/detail/*` in find-and-replace, it starts searching these ignored folders again. :confused:

**Comment by joseramonc** [comment]
Agree, it's happening to me too, It would be nice to have an option to ignore files in global search like fuzzy-finder does.

![screen shot 2016-12-09 at 11 50 41 am](https://cloud.githubusercontent.com/assets/1993929/21058898/bb19c1fe-be05-11e6-8de6-f7beb3c6b93e.png)

**Comment by chrishandorf** [comment]
Me too.  I don't want to search the node_modules subfolder when searching within my project.

**Comment by dpehrson** [comment]
Also hitting this. I'm confused why find-and-replace searches VCS ignored folders even though that config specifically says it won't?

**Comment by dwelle** [comment]
The `Ignored names` setting doesn't say it will be respected by `find-and-replace` package.

WTBS, it'd be nice to have a per-project settings of ignored paths for `find-and-replace`, possibly specified in a special file at the root of the project (I specifically _don't want_ to put the paths into `.gitignore`).
